workflows/flows/import_genome_assembly_links.py

```diff
@@ -0,0 +1,303 @@
+                f"Missing required columns in TSV: {', '.join(missing_columns)}"
+            )
+            raise ValueError(
+                f"Missing required columns in TSV: {', '.join(missing_columns)}"
+            )
+
+        for row in reader:
+            records.append(row)
+
+    logger.info(f"Read {len(records)} records from TSV file")
+    return records
+
+
+@task(name="Process TSV records")
+def process_tsv_records(records: List[Dict[str, str]]) -> List[Dict[str, str]]:
+    """
+    Processes the TSV records, validating and cleaning the data.
+
+    Args:
+        records: List of dictionaries, one for each row in the TSV file
+
+    Returns:
+        List of validated and cleaned records
+    """
+    logger = get_run_logger()
+    validated_records = []
+
+    for i, record in enumerate(records, 1):
+        # Check for required fields
+        if not record.get("primary_assembly"):
+            logger.warning(f"Row {i}: Missing primary_assembly, skipping")
+            continue
+
+        if not record.get("genome"):
+            logger.warning(f"Row {i}: Missing genome, skipping")
+            continue
+
+        # Clean up the record
+        cleaned_record = {
+            "primary_assembly": record["primary_assembly"].strip(),
+            "genome": record["genome"].strip(),
+            "mag_accession": (
+                record.get("mag_accession", "").strip()
+                if record.get("mag_accession")
+                else None
+            ),
+            "species_rep": (
+                record.get("species_rep", "").strip()
+                if record.get("species_rep")
+                else None
+            ),
+        }
+
+        validated_records.append(cleaned_record)
+
+    logger.info(f"Validated {len(validated_records)} records")
+    return validated_records
+
+
+@task(name="Find Genome and Assembly objects")
+def find_objects(
+    records: List[Dict[str, str]]
+) -> List[Tuple[Dict[str, str], Optional[Genome], Optional[Assembly]]]:
+    """
+    Finds the Genome and Assembly objects for each record.
+
+    Args:
+        records: List of validated and cleaned records
+
+    Returns:
+        List of tuples containing the record, Genome object, and Assembly object
+    """
+    logger = get_run_logger()
+    results = []
+
+    for record in records:
+        genome = None
+        assembly = None
+
+        # Find the Genome
+        try:
+            genome = Genome.objects.get(accession=record["genome"])
+        except ObjectDoesNotExist:
+            logger.warning(f"Genome not found: {record['genome']}")
+
+        # Find the Assembly
+        try:
+            assembly = Assembly.objects.get_by_accession(record["primary_assembly"])
+        except (ObjectDoesNotExist, Assembly.MultipleObjectsReturned):
+            logger.warning(
+                f"Assembly not found or multiple found: {record['primary_assembly']}"
+            )
+
+        results.append((record, genome, assembly))
```

Comment:
This follows Sandy comment about performance, this is likely to make loads of queries. I think we need to batch these with `IN` instead.